### ISSUE TITLE
Add a toggle to disable the injecting services feature

### ISSUE THREAD

**Issue by pkly** [Haehnchen/idea-php-symfony2-plugin#2397]
It's bad for me as I know what I want to write and I have 5 suggestions that I don't need. It keeps adding services to my classes and it's annoying. I can't find a way to disable it atm, is there one?

**Comment by kAlvaro** [comment]
When I use auto-completion for `$this->`, the plug-in is adding non-existing class properties with top priority and, should I pick one, it changes or creates the class constructor to inject a new service. IMHO, this is wrong for several reasons:

- It can break my app by introducing circular dependencies.
- I don't want the IDE to make design decisions for me, let alone without telling me.
- 99% of the times I type `$this->` what I want is to use existing class properties, not add new ones.
- When you want to add new properties, the intuitive way is to add a new property directly to either the class or the constructor. Doing it from the location where you want to use and as part of auto-completion is a bit convoluted.

Is there a way to disable this?

![image](https://github.com/user-attachments/assets/d8bdd4fe-ae1f-4cd2-a9e5-20b9d41ec439)

**Comment by pkly** [comment]
yeah it's annoying and barely works, doesn't respect formatting styles and is overall not a good thing

**Comment by pkly** [comment]
@Haehnchen any chance this could be added as a toggle in a future update? It keeps happening and breaking my working code.
